server/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/delegates/reader/CobolLineReaderImpl.java

```diff
@@ -151,7 +170,51 @@ public CobolLineReaderImpl(CobolLineReaderDelegate delegate, MessageService mess
                             INDICATOR_AREA_INDEX + 1))));
   }
 
+  @NonNull
+  private Optional<SyntaxError> checkLineLength(
+      @NonNull String line, @NonNull String uri, int lineNumber) {
+    if (line.length() <= 80) {
+      return Optional.empty();
+    }
+    return Optional.of(
+        createError(
+            uri,
+            messageService.getMessage("CobolLineReaderImpl.longLineMsg"),
+            lineNumber,
+            MAX_LINE_LENGTH,
+            line.length()));
+  }
+
+  private Optional<SyntaxError> checkSequenceArea(
+      @NonNull String line, @NonNull String uri, int lineNumber, int contentStart) {
+    if (isSequenceNumberFormatCorrect(line, contentStart)) {
+      return Optional.empty();
+    }
+    return Optional.of(
+        createError(
+            uri,
+            messageService.getMessage("CompilerDirectivesTransformation.sequenceNumber"),
+            lineNumber,
+            0,
+            1));
+  }
+
+  private boolean isSequenceNumberFormatCorrect(String line, int contentStart) {
+    // issue error the sequence must start with a number.
+    return contentStart < ProcessingConstants.INDICATOR_AREA
+        || StringUtils.isBlank(line.substring(0, contentStart))
+        || Character.isDigit(line.charAt(0));
+  }
+
+  private int getLineContentStart(String line) {
```

Comment:
Instead of it, you can change your `COMPILER_DIRECTIVE_LINE` to `"^(?i)\\s*(?<codeLine>(CBL|PROCESS) .+)"`. And extract line without spaces via `matcher.group("codeLine")`